### PR TITLE
Fix some of the RuboCop RSpec TODO items

### PR DESCRIPTION
**Pull request by Drenmi**
As a continuation to @backus' great work in #4904. This fixes some more RSpec related items in `.rubocop_todo.yml`, and introduces `#expect_offense` and `#expect_no_offenses` in some places.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests(`rake spec`) are passing.
* [X] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by Drenmi** [comment]
The CI failure is coming from the intermittent issue that appeared after we started running parallel tests:

```ruby
NameError:
 uninitialized constant RSpec::Support::Differ
```

**Comment by backus** [comment]
@Drenmi these changes look great. Kudos

File: spec/rubocop/cop/lint/unneeded_disable_spec.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   describe '.check' do
     let(:cop) do
       cop = described_class.new
-      cop.instance_eval { @options[:auto_correct] = true }
+      cop.instance_variable_get(:@options)[:auto_correct] = true
```
Comment (backus) [inline spec/rubocop/cop/lint/unneeded_disable_spec.rb]: I have nothing wrong with the test change here, but from a design perspective is it possible to set auto_correct in a less invasive way?

Comment (Drenmi) [reply spec/rubocop/cop/lint/unneeded_disable_spec.rb]: I tried a few different ways, but couldn't find any. 😕 Will need to look into the implementation, and why this cop is seemingly a special case.

**Comment by Drenmi** [comment]
@bbatsov: Rebased.

**Pull request merged**